Userland/Utilities/grep.cpp

```diff
@@ -195,25 +197,28 @@ ErrorOr<int> serenity_main(Main::Arguments args)
 
         auto handle_file = [&matches, binary_mode, suppress_errors, count_lines, quiet_mode,
                                user_specified_multiple_files, &matched_line_count, &did_match_something](StringView filename, bool print_filename) -> bool {
-            auto file = Core::File::construct(filename);
-            if (!file->open(Core::OpenMode::ReadOnly)) {
+            auto file_or_error = Core::Stream::File::open(filename, Core::Stream::OpenMode::Read);
+            if (file_or_error.is_error()) {
                 if (!suppress_errors)
-                    warnln("Failed to open {}: {}", filename, file->error_string());
+                    warnln("Failed to open {}: {}", filename, file_or_error.release_error());
                 return false;
             }
 
-            auto file_size_or_error = Core::File::size(filename);
-            if (file_size_or_error.is_error()) {
-                if (!suppress_errors)
-                    warnln("Failed to retrieve size of {}: {}", filename, strerror(file_size_or_error.error().code()));
-                return false;
-            }
+            auto file = file_or_error.release_value();
+            auto buffered_file = Core::Stream::BufferedFile::create(move(file)).release_value_but_fixme_should_propagate_errors();
 
-            auto file_size = file_size_or_error.release_value();
+            for (size_t line_number = 1; buffered_file->can_read_line().release_value_but_fixme_should_propagate_errors(); ++line_number) {
+                auto buffer = FixedArray<u8>::must_create_but_fixme_should_propagate_errors(PAGE_SIZE);
```

Comment:
Can't we use the normal `Array` here instead of `FixedArray`? If I recall correctly, the latter has some heavy performance implications when called in a loop like this. Considering that we don't compile our `grep` for Lagom, I don't think `PAGE_SIZE` not being `constexpr` on some platforms would be an issue either.